### ISSUE TITLE
mime types are not deleted

### ISSUE THREAD

**Issue by neilfrawley** [chef-cookbooks/iis#321]
### Cookbook version
iis 5.1.0

### Chef-client version
12.19.36

### Platform Details
Windows Server 2012 R2

### Scenario:
When trying to use the _iis_root_ resource and the _delete_mime_maps_ property the mime types do not get deleted, but the Chef run reports no error.

### Steps to Reproduce:
In metadata.rb in a cookbook specify;
depends 'iis', '~> 5.1.0'

In a recipe specify this code;
```ruby
iis_root 'remove mime types' do
  action :delete
  delete_mime_maps [
    "fileExtension='.rpm',mimeType='audio/x-pn-realaudio-plugin'",
    "fileExtension='.msi',mimeType='application/octet-stream'"
  ]
end
```

### Expected Result:
The 2 mime types get deleted from IIS.

### Actual Result:
The mime types do not get deleted from IIS. The Chef run in test kitchen reports as (up to date), yet the mime types are still present.

**Comment by EasyAsABC123** [comment]
Creating inspec tests for this so we can validate the code and fix if necessary.  Sorry for the delay in communication

**Comment by neilfrawley** [comment]
Thanks. Let me know if you need anything else from me. Cheers

**Comment by EasyAsABC123** [comment]
We are in the process of re-writing all the resources (moved to doing one at a time)...example: https://github.com/chef-cookbooks/iis/pull/322 i'll be updating them alphabetically so once we get to iis_root this will be re-written to a custom resource and will have actual tests.  any bugs (including this one) will be resolved then.

**Comment by EasyAsABC123** [comment]
@neilfrawley i think this is because of the section lock, while re-writing the iis_module resource i've run across something similar.

**Comment by EasyAsABC123** [comment]
I'll add a test for this when i get to the `iis_root` resource (next on the list)

**Comment by neilfrawley** [comment]
OK fantastic, thanks

**Comment by EasyAsABC123** [comment]
@neilfrawley resolved in #351 according to the test
